### DISCUSSION TITLE
OpenHASP & MQTT over TLS (Emqx Cloud)

### DISCUSSION THREAD

**Discussion by AdrienV** [HASwitchPlate/openHASP#547] [Q&A]
Hi guys, 

I am trying to configure and use emqx cloud on a screen with openHasp. I have some datas from other iot products sent to topics on Emqx cloud, and I would like to get these datas on my screen (not sure if Mqtt is the good way to do that).

I tried to configure the Mqtt on my openHasp, but it seems to be impossible because I can't add a certificate or fingerprint to connect over TLS. 
<img width="671" alt="Capture d’écran 2023-09-24 à 11 28 08" src="https://github.com/HASwitchPlate/openHASP/assets/3740879/6a279b78-b52c-4350-bf1e-bbbedcbfe058">
<img width="609" alt="Capture d’écran 2023-09-24 à 11 28 46" src="https://github.com/HASwitchPlate/openHASP/assets/3740879/7709210e-633e-42a4-b0a8-854ceecae027">

I tried the configuration with their public Mqtt and the connection is successful, but I don't find any topic to help to connect over TLS with a certificate. 
<img width="690" alt="Capture d’écran 2023-09-24 à 11 37 25" src="https://github.com/HASwitchPlate/openHASP/assets/3740879/415fb300-c595-4a17-a806-da47dda682ab">
<img width="550" alt="Capture d’écran 2023-09-24 à 11 40 20" src="https://github.com/HASwitchPlate/openHASP/assets/3740879/806c2455-645a-4a71-bf95-08e1bbadc27c">

Any one know how to do please ?

Thanks for your help :)

**Comment by fvanroie** [comment]
There is currently no support for MQTT over TLS in openHASP.
The [MQTT client](https://github.com/HASwitchPlate/openHASP/blob/f28627c14d73bf23cc3a768743b929dbc0546d97/src/mqtt/hasp_mqtt_esp.cpp#L555-L569) needs to be adapted and tested to allow these configurations.

**Comment by darkman7hr** [comment]
+1
